server.js

```diff
@@ -19,42 +19,49 @@ app.use(cors()); //enables server to talk to local things
 
 const client = new pg.Client(DATABASE_URL);
 client.on('error', (error) => console.error(error))
-//========================= Routes =========================================
+
+//========================= Routes =========================================================================================
 app.get('/location', sendLocationData)
+app.get('/weather', sendWeatherData);
+app.get('/trails', sendTrailData);
+// app.get('/movies', sendMoviesData);
+// app.get('/yelp', sendYelpData);
+
+//========================= Route Handlers =====================================================================================
 function sendLocationData(request, response){
-  const SQLStatement = 'SELECT * FROM locations;';
-    client.query(SQLStatement)
-      .then(resultFromSQL =>{
-        if(resultFromSQL.rowCount > 0){
-          response.send(resultFromSQL.rows[0])
-        }else{
-          const thingToSearchFor = request.query.city;
-          const urlToSearch = `https://us1.locationiq.com/v1/search.php?key=${GEOCODE_API_KEY}&q=${thingToSearchFor}&format=json`;
-        
-          superagent.get(urlToSearch)
-            .then(whateverComesBack => {
-              const superagentResultArray = whateverComesBack.body;
-              const constructedLocations = new Locations(superagentResultArray)
-              const queryString = 'INSERT INTO locations(search_query, latitude, longitude, formatted_query) VALUES ($1, $2, $3, $4)';
-              const valueArray = [constructedLocations.search_query, constructedLocations.latitude, constructedLocations.longitude, constructedLocations.formatted_query];
-              client.query(queryString, valueArray)
-                .then(()=>{
-                  response.status(201).send(constructedLocations)
-                })
-            })
-            .catch(error => {
-              console.log(error);
-              response.status(500).send(error.message);
-          });
-        }
-      });
+  // const SQLStatement = 'SELECT * FROM locations'
+  const thingToSearchFor = request.query.city;
+  const urlToSearch = `https://us1.locationiq.com/v1/search.php?key=${GEOCODE_API_KEY}&q=${thingToSearchFor}&format=json`;
+
+  client.query( 'SELECT * FROM locations WHERE search_query=$1', [thingToSearchFor])
+    .then(resultFromSQL =>{
+      if(resultFromSQL.rowCount === 1){
+        response.send(resultFromSQL.rows[0])
+      }else{
+        superagent.get(urlToSearch)
+          .then(whateverComesBack => {
+            const superagentResultArray = whateverComesBack.body;
+            const constructedLocations = new Locations(superagentResultArray)

```

Comment:
Nice use of the constructor before storing in the DB and sending back to the client. Good flow.